src/client/game.cpp

```diff
@@ -2517,23 +2488,34 @@ void Game::updatePlayerControl(const CameraOrientation &cam)
 	 */
 	if (m_cache_hold_aux1) {
 		control.aux1 = control.aux1 ^ true;
-		keypress_bits ^= ((u32)(1U << 5));
 	}
 #endif
 
-	LocalPlayer *player = client->getEnv().getLocalPlayer();
+	u32 keypress_bits = (
+			( (u32)(control.jump  & 0x1) << 4) |
+			( (u32)(control.aux1  & 0x1) << 5) |
+			( (u32)(control.sneak & 0x1) << 6) |
+			( (u32)(control.dig   & 0x1) << 7) |
+			( (u32)(control.place & 0x1) << 8) |
+			( (u32)(control.zoom  & 0x1) << 9)
+		);
 
-	// autojump if set: simulate "jump" key
-	if (player->getAutojump()) {
-		control.jump = true;
-		keypress_bits |= 1U << 4;
-	}
+	// Legacy fields to ensure mod compatibility
+	if (control.movement_speed > 0.001f) {
+		float absolute_direction;
 
-	// autoforward if set: simulate "up" key
-	if (player->getPlayerSettings().continuous_forward &&
-			client->activeObjectsReceived() && !player->isDead()) {
-		control.up = true;
-		keypress_bits |= 1U << 0;
+		// Check in original orientation (absolute value indicates forward / backward)
+		absolute_direction = abs(control.movement_direction);
+		keypress_bits |= (u32)(((absolute_direction < ((3.0 * M_PI) / 8.0)) & 0x1) << 0); // Forward
```

Comment:
readability, I suggest doing these for the others too
```suggestion
		if (absolute_direction < (3.0f / 8.0f * M_PI))
			keypress_bits |= (u32)(0x1 << 0); // Forward
```